XCUITests/SearchTest.swift

```diff
@@ -0,0 +1,167 @@
+        
+        app.buttons["No"].tap()
+        waitforNoExistence(app.tables["SiteTable"].buttons[SuggestedSite])
+        
+        // Verify that it is possible to enable suggestions after selecting No
+        app.buttons["Cancel"].tap()
+        suggestionsOnOff()
+        typeOnSearchBar(text: "foobar")
+        waitforExistence(app.tables["SiteTable"].buttons[SuggestedSite])
+    }
+    
+    func testDoNotShowSuggestionsWhenEnteringURL() {
+        // According to bug 1192155 if a string contains /, do not show suggestions, if there a space an a string, the suggestions are shown again
+        typeOnSearchBar(text: "foobar")
+        waitforExistence(app.staticTexts[LabelPrompt])
+        
+        // No suggestions should be shown
+        waitforNoExistence(app.tables["SiteTable"].buttons[SuggestedSite])
+        
+        // Enable Search suggestion
+        app.buttons["Yes"].tap()
+        
+        // Suggestions should be shown
+        waitforExistence(app.tables["SiteTable"].buttons[SuggestedSite])
+        
+        // Typing / should stop showing suggestions
+        app.textFields["address"].typeText("/")
+        waitforNoExistence(app.tables["SiteTable"].buttons[SuggestedSite])
+        
+        // Typing space and char after / should show suggestions again
+        app.textFields["address"].typeText(" b")
+        waitforExistence(app.tables["SiteTable"].buttons["foobar burn cd"])
+    }
+    
+    func testCopyPasteComplete() {
+        // Copy, Paste and Go to url
+        typeOnSearchBar(text: "www.mozilla.org")
+        app.textFields["address"].press(forDuration: 5)
+        app.menuItems["Select All"].tap()
+        app.menuItems["Copy"].tap()
+        
+        app.buttons["Cancel"].tap()
+        
+        app.textFields["url"].tap()
+        app.textFields["address"].tap()
+        app.menuItems["Paste"].tap()
+        
+        // Verify that the Paste shows the search controller with prompt
+        waitforExistence(app.staticTexts[LabelPrompt])
+        app.typeText("\r")
+        
+        // Check that the website is loaded
+        waitForValueContains(app.textFields["url"], value: "https://www.mozilla.org/")
+        
+        // Go back, write part of moz, check the autocompletion
+        app.buttons["TabToolbar.backButton"].tap()
+        typeOnSearchBar(text: "moz")
+        
+        waitForValueContains(app.textFields["address"], value: "mozilla.org")
+        let value = app.textFields["address"].value
+        XCTAssertEqual(value as? String, "mozilla.org/")
+    }
+    
+    private func changeSearchEngine(searchEngine: String) {
+        let tablesQuery2 = app.tables
+        tablesQuery2.staticTexts[searchEngine].tap()
+        
+        app.navigationBars["Search"].buttons["Settings"].tap()
+        app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
+        typeOnSearchBar(text: "foo\n")
```

Comment:
here I think the test can be simplified by using  navigator.openNewURL(urlString: "foo") instead of 3 lines above